### PR TITLE
refactor: Optimize Dockerfiles a bit

### PR DESCRIPTION
**Pull request by sjinks**
* Optimize Dockerfiles a bit by removing redundant commands.
 * Install `git` into `php-fpm` images to upport Codespaces.
 * Install `composer`.

### CODE REVIEW

File: mu-plugins/Dockerfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 FROM ghcr.io/automattic/vip-container-images/alpine:3.17.1@sha256:fbd2cb3cd937080dd4a701d0433e7b0604260e3542a805cef815790e2f921ebf
 
 RUN \
-    apk add --no-cache git shadow && \
+    apk add --no-cache git && \
```
Comment (sjinks) [inline mu-plugins/Dockerfile:4]: `shadow` comes from `ghcr.io/automattic/vip-container-images/alpine:3.17.1`

File: nginx/Dockerfile [hunk 2]
```diff
@@ -1,7 +1,5 @@
 FROM nginx:1.23.3-alpine@sha256:659610aadb34b7967dea7686926fdcf08d588a71c5121edb094ce0e4cdbc45e6
 
-RUN apk add --no-cache openssl shadow
+RUN apk add --no-cache shadow
```
Comment (sjinks) [inline nginx/Dockerfile:3]: We don't use `openssl` here

File: nginx/Dockerfile [hunk 3]
```diff
@@ -1,7 +1,5 @@
...
 
 COPY nginx/default.conf /etc/nginx/conf.d/default.conf
-
-CMD ["nginx", "-g", "daemon off;"]
```
Comment (sjinks) [inline nginx/Dockerfile:7]: This `CMD` is already present in the parent image.

File: wordpress/Dockerfile [hunk 4]
```diff
@@ -18,5 +18,4 @@ RUN \
 COPY extra/ /wordpress/wordpress/
 
 FROM ghcr.io/automattic/vip-container-images/alpine:3.17.1@sha256:fbd2cb3cd937080dd4a701d0433e7b0604260e3542a805cef815790e2f921ebf
-RUN apk add --no-cache shadow
```
Comment (sjinks) [inline wordpress/Dockerfile:21]: `shadow` is already in `ghcr.io/automattic/vip-container-images/alpine:3.17.1`

File: php-fpm/Dockerfile.80 [hunk 5]
```diff
@@ -2,7 +2,7 @@ FROM alpine:3.16.3@sha256:b95359c2505145f16c6aa384f9cc74eeff78eb36d308ca4fd902ee
 
 RUN \
     echo "@testing https://dl-cdn.alpinelinux.org/alpine/edge/testing" >> /etc/apk/repositories && \
-    apk add --no-cache ca-certificates curl tar xz openssl less bash mariadb-client su-exec shadow jq icu-data-full php8 php8-fpm php8-pear \
+    apk add --no-cache ca-certificates curl tar xz openssl less bash mariadb-client su-exec shadow git jq icu-data-full php8 php8-fpm php8-pear \
```
Comment (sjinks) [inline php-fpm/Dockerfile.80:5]: Codespaces will complain without `git`.

**Review by chriszarate — APPROVED** [review]

**Pull request merged**